src/app/product/product.component.ts

```diff
@@ -24,14 +38,23 @@ export class ProductComponent implements OnInit {
       .pipe(
         debounceTime(1000),
```

Comment:
Dodaj: `takeUntil(this.unsubscribeSubject$)`

Zatim:
1. Definises na nivou komponente: `unsubscribeSubject$ = new Subject<void>();`
2. U ngOnDestroy() metodi odradis:
```
this.unsubscribeSubject$.next();
this.unsubscribeSubject$.complete();
```